### ISSUE TITLE
Cant Add Account 1.41.1 and 1.41.2

### ISSUE THREAD

**Issue by ncapito** [microsoft/azuredatastudio#22227]
I was having issues this morning with my Active Directory Universal MFA so i "removed the account."  It now wont let me add it back.  I've tried newer versions of ADS, I've restarted, i've cleared out he Library/Application Support/azuredatastudio directory.  Nothing seems to work.

Version info:
Version: 1.41.2 (Universal)
Commit: 0f0960d0e774e1ce903b79378addff96117c27a6
Date: 2023-02-08T20:15:24.215Z (3 wks ago)
VS Code: 1.67.0
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Darwin x64 22.3.0

When i go to Linked Accounts -> Add an account i get errors with: "TypeError: Cannot read properties of undefined (reading 'authLibrary')" 

![image](https://user-images.githubusercontent.com/601055/223785798-d0ceb758-5054-4daf-aa56-1416abf84fd3.png)

>notificationsAlerts.ts:42 error
onDidChangeNotification @ notificationsAlerts.ts:42
(anonymous) @ notificationsAlerts.ts:28
invoke @ event.ts:575
deliver @ event.ts:779
fire @ event.ts:740
addNotification @ notifications.ts:206
notify @ notificationService.ts:137
(anonymous) @ mainThreadMessageService.ts:86
ZoneAwarePromise @ /Applications/Azure Data Studio.app/Contents/Resources/app/node_modules.asar/zone.js/dist/zone.js:1351
_showMessage @ mainThreadMessageService.ts:44
$showMessage @ mainThreadMessageService.ts:38
_doInvokeHandler @ rpcProtocol.ts:475
_invokeHandler @ rpcProtocol.ts:460
_receiveRequest @ rpcProtocol.ts:374
_receiveOneMessage @ rpcProtocol.ts:296
(anonymous) @ rpcProtocol.ts:161
invoke @ event.ts:575
deliver @ event.ts:779
fire @ event.ts:740
fire @ ipc.net.ts:638
_receiveMessage @ ipc.net.ts:958
(anonymous) @ ipc.net.ts:831
invoke @ event.ts:575
deliver @ event.ts:779
fire @ event.ts:740
acceptChunk @ ipc.net.ts:382
(anonymous) @ ipc.net.ts:338
L @ ipc.net.ts:60
emit @ node:events:526
addChunk @ node:internal/streams/readable:315
readableAddChunk @ node:internal/streams/readable:289
Readable.push @ node:internal/streams/readable:228
onStreamRead @ node:internal/stream_base_commons:190
Show 3 more frames
log.ts:313   ERR TypeError: Cannot read properties of undefined (reading 'providerId')
    at Pi.findAccountByKey (accountStore.ts:105:15)
    at accountStore.ts:31:63
    at Array.findIndex (<anonymous>)
    at accountStore.ts:31:29
    at _ZoneDelegate.invoke (/Applications/Azure Data Studio.app/Contents/Resources/app/node_modules.asar/zone.js/dist/zone.js:409:30)
    at Zone.run (/Applications/Azure Data Studio.app/Contents/Resources/app/node_modules.asar/zone.js/dist/zone.js:169:47)
    at /Applications/Azure Data Studio.app/Contents/Resources/app/node_modules.asar/zone.js/dist/zone.js:1326:38
    at _ZoneDelegate.invokeTask (/Applications/Azure Data Studio.app/Contents/Resources/app/node_modules.asar/zone.js/dist/zone.js:443:35)
    at Zone.runTask (/Applications/Azure Data Studio.app/Contents/Resources/app/node_modules.asar/zone.js/dist/zone.js:214:51)
    at drainMicroTaskQueue (/Applications/Azure Data Studio.app/Contents/Resources/app/node_modules.asar/zone.js/dist/zone.js:632:39)
    at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
log.ts:313   ERR adding account failed
log.ts:313   ERR Error while adding account: Error: Adding account failed, check Azure Accounts log for more info.

**Comment by cheenamalhotra** [comment]
This looks related to the recent bug we fixed here: https://github.com/microsoft/azuredatastudio/pull/22140

@ncapito 

Can you add this setting in settings.json and try again?

```json
    "azure.authenticationLibrary": "MSAL"
```

**Comment by ncapito** [comment]
@cheenamalhotra  It's currently set to MSAL (when i try with this setting i get the error above). When i toggled to ADAP it opened chrome and then locked up waiting https://login.microsoftonline.com/common/oauth2/authorize?response_type=code&response_mode=query&client_id=removed&redirect_uri=http%3A%2F%2Flocalhost%3A49783%2Fredirect&state=49783%2CcGt742e0w%252BjfKr2Owo619Q%253D%253D&prompt=select_account&code_challenge_method=S256&code_challenge=removed&resource=https%3A%2F%2Fmanagement.core.windows.net%2F

**Comment by cheenamalhotra** [comment]
It may not be set in the settings.json as it's default. Please try adding explicitly by opening settings.json:
1. Open command pallete (Ctrl+Shift+P)
2. Run command: Preferences: Open Default Settings (JSON) (workbench.action.openRawDefaultSettings)

> When i toggled to ADAP it opened chrome and then locked up waiting...

Are you in proxy enabled env by any chance? If yes, you may want to configure proxy in http.proxy setting:

```json
"http.proxy": "https://userName@fqdn:yourPassword@yourProxyURL.com:8080"
```

**Comment by ncapito** [comment]
@cheenamalhotra  I'm not in a proxy enabled environment.

I edited

1. settings.json and added   "azure.authenticationLibrary": "MSAL"
2. restarted ADS
3. tried to connect account and got an error but the message was different.
![image](https://user-images.githubusercontent.com/601055/223839833-3318ccd9-56e3-42ec-b27e-08bc2344709e.png)


![image](https://user-images.githubusercontent.com/601055/223839672-2e5aecb2-5765-49ff-894c-0be07badcae9.png)



I did check with my colleagues.  They are all on 141.2 but dont have an issue because they did not remove the account.

**Comment by ncapito** [comment]
@cheenamalhotra  I have an update on this one.  I was able to get ADAL returning and throwing the same error now.  My firewall was blocking access to Azure Data Studio (since i removed the app and reinstalled it).  Once I allowed connections i now get the same error:

![image](https://user-images.githubusercontent.com/601055/223847868-4d052e35-6073-4519-8d53-a9c023f040b5.png)

the console had this in it:

![image](https://user-images.githubusercontent.com/601055/223847946-94ed9dda-0bb9-4969-9b5b-68ecf8f45aad.png)

**Comment by cpirtea-bun** [comment]
I have the same problem. Went as far as remove all traces of Azure Data Studio and reinstalled clean. Nothing works. MSAL or ADAL both end up in the same 2 errors:
"TypeError: Cannot read properties of undefined (reading 'authLibrary')"
"Error: Adding account failed, check Azure Accounts log for more info."

**Comment by cheenamalhotra** [comment]
@ncapito

Can you please fetch Azure account logs as per above comment? 

@cpirtea-bun 

Please verify if you've added this in settings.json (as I mentioned above: https://github.com/microsoft/azuredatastudio/issues/22227#issuecomment-1460715233, https://github.com/microsoft/azuredatastudio/issues/22227#issuecomment-1460610800
If that's done and issue persists, please enable detailed logs and upload logs here.

**Comment by cpirtea-bun** [comment]
@cheenamalhotra will upload logs.
Just wanted to mention I installed the insider build and same outcome, however if I switch back to ADAL I get: Error: certificate has expired

**Comment by eduardofontes** [comment]
Hi Everyone!

Same problem here!

**Comment by eduardofontes** [comment]
[Error]: MSAL: Error requesting auth code - [{"errorCode":"endpoints_resolution_error","errorMessage":"Error: could not resolve endpoints. Please check network and try again. Detail: ClientAuthError: openid_config_error: Could not retrieve endpoints. Check your authority and verify the .well-known/openid-configuration endpoint returns the required endpoints. Attempted to retrieve endpoints from: https://undefined/organizations/v2.0/.well-known/openid-configuration","subError":"","name":"ClientAuthError"}]

**Comment by cpirtea-bun** [comment]
css/common.css:1          Failed to load resource: net::ERR_FILE_NOT_FOUND
log.ts:313   ERR [Microsoft.mssql]: Menu item references a command `mssql.newServerRole` which is not defined in the 'commands' section.
log.ts:313   ERR [Microsoft.mssql]: Menu item references a command `mssql.newDatabaseRole` which is not defined in the 'commands' section.
log.ts:313   ERR [Microsoft.mssql]: Menu item references a command `mssql.newApplicationRole` which is not defined in the 'commands' section.
console.ts:137 [Extension Host] Initialized Azure account extension storage. (at c:\Users\cpirtea\AppData\Local\Programs\Azure Data Studio - Insiders\resources\app\extensions\azurecore\dist\extension.js:2:904820)
console.ts:137 [Extension Host] logFile for MicrosoftSqlToolsServiceLayer.exe is c:\Users\cpirtea\AppData\Roaming\azuredatastudio\logs\20230309T091345\exthost1\Microsoft.mssql\sqltools.log (at t.getCommonLaunchArgsAndCleanupOldLogFiles (c:\Users\cpirtea\AppData\Local\Programs\Azure Data Studio - Insiders\resources\app\extensions\mssql\dist\main.js:2:1072336))
console.ts:137 [Extension Host] This process (ui Extension Host) is pid: 5792 (at t.getCommonLaunchArgsAndCleanupOldLogFiles (c:\Users\cpirtea\AppData\Local\Programs\Azure Data Studio - Insiders\resources\app\extensions\mssql\dist\main.js:2:1072388))
console.ts:137 [Extension Host] Old log files deletion report: {} (at t.getCommonLaunchArgsAndCleanupOldLogFiles (c:\Users\cpirtea\AppData\Local\Programs\Azure Data Studio - Insiders\resources\app\extensions\mssql\dist\main.js:2:1072477))
console.ts:137 [Extension Host] logFile for MicrosoftSqlToolsCredentials.exe is c:\Users\cpirtea\AppData\Roaming\azuredatastudio\logs\20230309T091345\exthost1\Microsoft.mssql\credentialstore.log (at t.getCommonLaunchArgsAndCleanupOldLogFiles (c:\Users\cpirtea\AppData\Local\Programs\Azure Data Studio - Insiders\resources\app\extensions\mssql\dist\main.js:2:1072336))
console.ts:137 [Extension Host] This process (ui Extension Host) is pid: 5792 (at t.getCommonLaunchArgsAndCleanupOldLogFiles (c:\Users\cpirtea\AppData\Local\Programs\Azure Data Studio - Insiders\resources\app\extensions\mssql\dist\main.js:2:1072388))
console.ts:137 [Extension Host] Old log files deletion report: {} (at t.getCommonLaunchArgsAndCleanupOldLogFiles (c:\Users\cpirtea\AppData\Local\Programs\Azure Data Studio - Insiders\resources\app\extensions\mssql\dist\main.js:2:1072477))
console.ts:137 [Extension Host] logFile for SqlToolsResourceProviderService.exe is c:\Users\cpirtea\AppData\Roaming\azuredatastudio\logs\20230309T091345\exthost1\Microsoft.mssql\resourceprovider.log (at t.getCommonLaunchArgsAndCleanupOldLogFiles (c:\Users\cpirtea\AppData\Local\Programs\Azure Data Studio - Insiders\resources\app\extensions\mssql\dist\main.js:2:1072336))
console.ts:137 [Extension Host] This process (ui Extension Host) is pid: 5792 (at t.getCommonLaunchArgsAndCleanupOldLogFiles (c:\Users\cpirtea\AppData\Local\Programs\Azure Data Studio - Insiders\resources\app\extensions\mssql\dist\main.js:2:1072388))
console.ts:137 [Extension Host] Old log files deletion report: {} (at t.getCommonLaunchArgsAndCleanupOldLogFiles (c:\Users\cpirtea\AppData\Local\Programs\Azure Data Studio - Insiders\resources\app\extensions\mssql\dist\main.js:2:1072477))
console.ts:137 [Extension Host] Initializing stored accounts [] (at t.AzureAccountProvider._initialize (c:\Users\cpirtea\AppData\Local\Programs\Azure Data Studio - Insiders\resources\app\extensions\azurecore\dist\extension.js:2:524730))
mainThreadExtensionService.ts:79 [[object Object]]Unexpected token < in JSON at position 0
$onExtensionRuntimeError @ mainThreadExtensionService.ts:79
mainThreadExtensionService.ts:80 SyntaxError: Unexpected token < in JSON at position 0
	at JSON.parse (<anonymous>)
	at IncomingMessage.<anonymous> (c:\Users\cpirtea\AppData\Local\Programs\Azure Data Studio - Insiders\resources\app\extensions\azurecore\node_modules\@azure\msal-node\dist\msal-node.cjs.development.js:321:22)
	at IncomingMessage.emit (node:events:538:35)
	at endReadableNT (node:internal/streams/readable:1345:12)
	at process.processTicksAndRejections (node:internal/process/task_queues:83:21)
$onExtensionRuntimeError @ mainThreadExtensionService.ts:80
mainThreadExtensionService.ts:79 [[object Object]]Unexpected token < in JSON at position 0
$onExtensionRuntimeError @ mainThreadExtensionService.ts:79
_doInvokeHandler @ rpcProtocol.ts:475
_invokeHandler @ rpcProtocol.ts:460
_receiveRequest @ rpcProtocol.ts:374
_receiveOneMessage @ rpcProtocol.ts:296
(anonymous) @ rpcProtocol.ts:161
invoke @ event.ts:575
deliver @ event.ts:779
fire @ event.ts:740
fire @ ipc.net.ts:638
_receiveMessage @ ipc.net.ts:958
(anonymous) @ ipc.net.ts:831
invoke @ event.ts:575
deliver @ event.ts:779
fire @ event.ts:740
acceptChunk @ ipc.net.ts:382
(anonymous) @ ipc.net.ts:338
L @ ipc.net.ts:60
emit @ node:events:526
addChunk @ node:internal/streams/readable:315
readableAddChunk @ node:internal/streams/readable:289
Readable.push @ node:internal/streams/readable:228
onStreamRead @ node:internal/stream_base_commons:190
mainThreadExtensionService.ts:80 SyntaxError: Unexpected token < in JSON at position 0
	at JSON.parse (<anonymous>)
	at IncomingMessage.<anonymous> (c:\Users\cpirtea\AppData\Local\Programs\Azure Data Studio - Insiders\resources\app\extensions\azurecore\node_modules\@azure\msal-node\dist\msal-node.cjs.development.js:321:22)
	at IncomingMessage.emit (node:events:538:35)
	at endReadableNT (node:internal/streams/readable:1345:12)
	at process.processTicksAndRejections (node:internal/process/task_queues:83:21)
$onExtensionRuntimeError @ mainThreadExtensionService.ts:80
_doInvokeHandler @ rpcProtocol.ts:475
_invokeHandler @ rpcProtocol.ts:460
_receiveRequest @ rpcProtocol.ts:374
_receiveOneMessage @ rpcProtocol.ts:296
(anonymous) @ rpcProtocol.ts:161
invoke @ event.ts:575
deliver @ event.ts:779
fire @ event.ts:740
fire @ ipc.net.ts:638
_receiveMessage @ ipc.net.ts:958
(anonymous) @ ipc.net.ts:831
invoke @ event.ts:575
deliver @ event.ts:779
fire @ event.ts:740
acceptChunk @ ipc.net.ts:382
(anonymous) @ ipc.net.ts:338
L @ ipc.net.ts:60
emit @ node:events:526
addChunk @ node:internal/streams/readable:315
readableAddChunk @ node:internal/streams/readable:289
Readable.push @ node:internal/streams/readable:228
onStreamRead @ node:internal/stream_base_commons:190

**Comment by cheenamalhotra** [comment]
Thank you for the logs.

```
[Error]: MSAL: Error requesting auth code - 
[{
    "errorCode":"endpoints_resolution_error",
    "errorMessage":"Error: could not resolve endpoints. Please check network and try again. 
        Detail: ClientAuthError: openid_config_error: Could not retrieve endpoints. Check your authority and verify the .well-known/openid-configuration endpoint returns the required endpoints. 
        Attempted to retrieve endpoints from: https://undefined/organizations/v2.0/.well-known/openid-configuration",
    "subError":"",
    "name":"ClientAuthError"
}]
```

About this error, I recall seeing that when I was implementing MSAL support in [vscode-mssql](https://github.com/microsoft/vscode-mssql) extension, that I also reported here: https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/4879 but hadn't seen that in ADS yet. Now it makes me think if it's related to some other factors too. I'm going to try updating to latest [MSAL.js v1.16.0](https://github.com/AzureAD/microsoft-authentication-library-for-js/releases/tag/msal-node-v1.16.0) if that helps resolve this issue, as it seems to contain a proper fix.

```
onExtensionRuntimeError @ mainThreadExtensionService.ts:79
mainThreadExtensionService.ts:80 SyntaxError: Unexpected token < in JSON at position 0
  at JSON.parse ()
  at IncomingMessage. (c:\Users\cpirtea\AppData\Local\Programs\Azure Data Studio - Insiders\resources\app\extensions\azurecore\node_modules@azure\msal-node\dist\msal-node.cjs.development.js:321:22)
  at IncomingMessage.emit (node:events:538:35)
  at endReadableNT (node:internal/streams/readable:1345:12)
  at process.processTicksAndRejections (node:internal/process/task_queues:83:21)
  $onExtensionRuntimeError @ mainThreadExtensionService.ts:80
```

This seems related to a corrupted cache file leading to runtime error in MSAL, can you please try deleting cache file:

`C:\Users\~\AppData\Roaming\azuredatastudio\Azure Accounts\azureTokenCacheMsal-azure_publicCloud` (Windows)
or `/Users/~/Library/Application Support/azuredatastudio/Azure Accounts/azureTokenCacheMsal-azure_publicCloud` (Mac OS)

**Comment by cpirtea-bun** [comment]
I deleted the cache and same error appeared

**Comment by cpirtea-bun** [comment]
Just looking at the code the error comes from parsing the response so it should have anything to do with the cache.

**Comment by cpirtea-bun** [comment]
After you update to latest MSAL do I wait for next insider build?

**Comment by cheenamalhotra** [comment]
Yes, it'll be available in the next update.

**Comment by ncapito** [comment]
> We need more info to debug your Azure Active Directory issue. If you could attach your logs to the issue (ensure no private data is in them), it would help us fix the issue much faster.

In case this is still needed.

Here was the first log from adding account using ADAL.
[2-Azure Accounts.log](https://github.com/microsoft/azuredatastudio/files/10930738/2-Azure.Accounts.log)

Here is the log from the MSAL
[2-Azure Accounts.log](https://github.com/microsoft/azuredatastudio/files/10930748/2-Azure.Accounts.log)

**Comment by StacyCMay** [comment]
I am also seeing this behaviour using Azure Data Studio v1.41.2.

**Comment by cheenamalhotra** [comment]
> Yes, it'll be available in the next update.

Just FYI, MSAL-Node update needs more work in insiders and has been postponed, but we're actively investigating on issues with the stable release.

**Comment by StacyCMay** [comment]
> Yes, it'll be available in the next update.

Hotfix or May release?

**Comment by cheenamalhotra** [comment]
Hi everyone,

MSAL should be working now, however, we're investigating ADAL related issues.

Would request you to please try adding accounts using the MSAL library and let us know if you're still blocked due to any reason. Please upload logs if you're still blocked, and we can take a look. We don't anticipate a hotfix for this issue, as this was related with the AAD backend.

**Comment by StacyCMay** [comment]
> Hi everyone,
> 
> MSAL should be working now, however, we're investigating ADAL related issues.
> 
> Would request you to please try adding accounts using the MSAL library and let us know if you're still blocked due to any reason. Please upload logs if you're still blocked, and we can take a look. We don't anticipate a hotfix for this issue, as this was related with the AAD backend.

Still blocked. Error output: 
```
[Error]: MSAL: Error requesting auth code - [{"errorCode":"endpoints_resolution_error","errorMessage":"Error: could not resolve endpoints. Please check network and try again. Detail: ClientAuthError: openid_config_error: Could not retrieve endpoints. Check your authority and verify the .well-known/openid-configuration endpoint returns the required endpoints. Attempted to retrieve endpoints from: https://undefined/organizations/v2.0/.well-known/openid-configuration","subError":"","name":"ClientAuthError"}]
[Error]: Login failed: Error: error - []
[Error]: {
  "originalMessage": "Error requesting auth code",
  "originalException": {
    "errorCode": "endpoints_resolution_error",
    "errorMessage": "Error: could not resolve endpoints. Please check network and try again. Detail: ClientAuthError: openid_config_error: Could not retrieve endpoints. Check your authority and verify the .well-known/openid-configuration endpoint returns the required endpoints. Attempted to retrieve endpoints from: https://undefined/organizations/v2.0/.well-known/openid-configuration",
    "subError": "",
    "name": "ClientAuthError"
  }
} - []
```

**Comment by cpirtea-bun** [comment]
[1-Azure Accounts.log](https://github.com/microsoft/azuredatastudio/files/10936534/1-Azure.Accounts.log)

**Comment by cheenamalhotra** [comment]
Could you try adding account with Device Code authentication method? 
Since this is an MSAL issue https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/4879 I'll resume on updating MSAL library.

<img src="https://user-images.githubusercontent.com/13396919/224173786-f9126d99-61d5-453f-918a-1d7186a45445.png" width=600 />

**Comment by StacyCMay** [comment]
> Could you try adding account with Device Code authentication method? Since this is an MSAL issue [AzureAD/microsoft-authentication-library-for-js#4879](https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/4879) I'll resume on updating MSAL library.
> 
> <img alt="" width="600" src="https://user-images.githubusercontent.com/13396919/224173786-f9126d99-61d5-453f-918a-1d7186a45445.png">

Still erroring: 
```
[Error]: Login failed: ClientAuthError: endpoints_resolution_error: Error: could not resolve endpoints. Please check network and try again. Detail: ClientAuthError: openid_config_error: Could not retrieve endpoints. Check your authority and verify the .well-known/openid-configuration endpoint returns the required endpoints. Attempted to retrieve endpoints from: https://undefined/organizations/v2.0/.well-known/openid-configuration - []
[Error]: ClientAuthError: endpoints_resolution_error: Error: could not resolve endpoints. Please check network and try again. Detail: ClientAuthError: openid_config_error: Could not retrieve endpoints. Check your authority and verify the .well-known/openid-configuration endpoint returns the required endpoints. Attempted to retrieve endpoints from: https://undefined/organizations/v2.0/.well-known/openid-configuration - []
```

**Comment by jonmarkp** [comment]
I am having the exact same issue. I tried all the above steps and nothing has worked. 

<img width="1017" alt="image" src="https://user-images.githubusercontent.com/40213625/224212799-9a5433c9-3293-4bd1-9e96-d195c3ed1b27.png">

Here is the azure accounts log:

[1-Azure Accounts.log](https://github.com/microsoft/azuredatastudio/files/10938115/1-Azure.Accounts.log)

I am on a Mac and by accident removed the credentials and not I cannot re-add. On my Windows side, I did not remove the credentials and I can access the database just fine.

**Comment by cheenamalhotra** [comment]
Hi @StacyCMay @jonmarkp @cpirtea-bun 

The latest insiders build now contains the latest version of msal-node npm package. Could you please give it a try?
https://github.com/microsoft/azuredatastudio#try-out-the-latest-insiders-build-from-main-branch

**Comment by StacyCMay** [comment]
> Hi @StacyCMay @jonmarkp @cpirtea-bun
> 
> The latest insiders build now contains the latest version of msal-node npm package. Could you please give it a try? https://github.com/microsoft/azuredatastudio#try-out-the-latest-insiders-build-from-main-branch

Same issue: 
![image](https://user-images.githubusercontent.com/47575658/224382830-d47612b7-364a-467f-afa6-35e7a82730f4.png)
```[Error]: Login failed: ClientAuthError: endpoints_resolution_error: Error: could not resolve endpoints. Please check network and try again. Detail: ClientAuthError: openid_config_error: Could not retrieve endpoints. Check your authority and verify the .well-known/openid-configuration endpoint returns the required endpoints. Attempted to retrieve endpoints from: https://undefined/organizations/v2.0/.well-known/openid-configuration - []
```

**Comment by StacyCMay** [comment]
> > Hi @StacyCMay @jonmarkp @cpirtea-bun
> > The latest insiders build now contains the latest version of msal-node npm package. Could you please give it a try? https://github.com/microsoft/azuredatastudio#try-out-the-latest-insiders-build-from-main-branch
> 
> Same issue: ![image](https://user-images.githubusercontent.com/47575658/224382830-d47612b7-364a-467f-afa6-35e7a82730f4.png) `[Error]: Login failed: ClientAuthError: endpoints_resolution_error: Error: could not resolve endpoints. Please check network and try again. Detail: ClientAuthError: openid_config_error: Could not retrieve endpoints. Check your authority and verify the .well-known/openid-configuration endpoint returns the required endpoints. Attempted to retrieve endpoints from: https://undefined/organizations/v2.0/.well-known/openid-configuration - [] `

Version I am using :
```
Version: 1.43.0-insider
Commit: c163830bc4611af6c01458e7f5bd8f2546c4f523
Date: 2023-03-10T06:30:09.688Z
VS Code: 1.67.0
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Darwin arm64 22.3.0
```

**Comment by cheenamalhotra** [comment]
Can you upload the complete verbose Azure Account log with latest version?

**Comment by StacyCMay** [comment]
> Can you upload the complete verbose Azure Account log with latest version - starting with `[Verbose]: Starting login - []`

[1-Azure.Accounts.log](https://github.com/microsoft/azuredatastudio/files/10944843/1-Azure.Accounts.log)

**Comment by cheenamalhotra** [comment]
It appears this is also related with the error reported above with ADAL:

```
[Error]: [Fri, 10 Mar 2023 17:26:49 GMT] : @azure/msal-node@1.16.0 : Error - A non-MSALJS error was thrown while attempting to get the cloud instance discovery metadata.
Error: Error
Error Description: Error: certificate has expired - []
```

I see similar issues reported here: 
https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/5779
https://github.com/OfficeDev/TeamsFx/issues/7995

They look beyond the scope of Azure Data Studio and are possibly related with Azure configurations.

I would recommend opening an Azure support ticket for the same with your account, as we cannot diagnose further on Azure account configurations from GitHub. It seems to be affecting a subset of customers, and not everyone. You may link this issue and other similar issues in the ticket for context.

**Comment by kburtram** [comment]
If anyone who is impacted by this issue is able to open an Azure support ticket please reference this issue and Azure Data Studio.  This will route the incident to us and allow us to look at backend service telemetry for the auth requests and engage with the AAD team.  Thanks!

> I would recommend opening an Azure support ticket for the same with your account, as we cannot diagnose further on Azure account configurations from GitHub. It seems to be affecting a subset of customers, and not everyone. You may link this issue and other similar issues in the ticket for context.

**Comment by ncapito** [comment]
I just re-ran the test with the new build and i can confirm i got a similar error.  Just to make sure I'm tracking you all want us to file a ticket in Azure Support about this issue.

My Error was: 

[Verbose]: MsalCachePluginProvider: Using cache path /Users/ncapito/Library/Application Support/azuredatastudio/Azure Accounts and serviceName azureTokenCacheMsal-azure_publicCloud - []
[Verbose]: Starting login - []
[Verbose]: Fetching token for tenant {0} - ["common"]
[Error]: Login failed: Error: certificate has expired - []
[Error]: Error: certificate has expired - []

**Comment by cssuh** [comment]
@ncapito Hi yes that's correct, please open an Azure Support ticket - our team (Azure Data Studio) is under Azure SQL DB.

**Comment by kburtram** [comment]
@ncapito if possible, it would be helpful to have an Azure Support case for this, so we can get details on the user identities and times of the auth attempts (to try to correlate with errors in the service telemetry).  And to be able to request more detailed logs that we can't collect over GitHub due to potential for PII data.

**Comment by cheenamalhotra** [comment]
Hi everyone,

I've learned from an internal thread, [v1.39.1](https://github.com/microsoft/azuredatastudio/releases/tag/1.39.1) of ADS seems to work in this use-case, can someone who's affected confirm the same? If it does, you may use the same to stay unblocked, while we continue with investigations.

**Comment by StacyCMay** [comment]
> Hi everyone,
> 
> I've learned from an internal thread, [v1.39.1](https://github.com/microsoft/azuredatastudio/releases/tag/1.39.1) of ADS seems to work in this use-case, can someone who's affected confirm the same? Also please try with [v1.40.2](https://github.com/microsoft/azuredatastudio/releases/tag/1.40.2) as that would be useful for us to diagnose the problem. If it does, you may use the same to stay unblocked, while we continue with investigations.

1.39.1 works, but 1.40.2 does not. The error logs from 1.40.2 are:
```
[Verbose]: Starting login - []
[Error]: Login failed - []
[Error]: Error: certificate has expired - []
```

**Comment by cheenamalhotra** [comment]
Hi everyone,

Would request you to try out the [VSCode-MSSQL](https://github.com/microsoft/vscode-mssql) extension: [v1.18.0 Release Candidate](https://azuredatastudiobuilds.blob.core.windows.net/test/vscode-mssql/mssql-1.18.0.vsix) to help us validate if Azure MFA authentication works here in your environment.

The extension supports MSAL (default) and ADAL library switching with configuration:
```
    "mssql.azureAuthenticationLibrary": "MSAL"
```
To connect, you can run command: `mssql.connect` or open a new Connection in 'Connections' view.

**Comment by StacyCMay** [comment]
I'm assuming you would like me to try this out with version 1.42.2 of ADS?

**Comment by StacyCMay** [comment]
Works like a charm!

**Comment by ncapito** [comment]
Quick update.  I did file a ticket with Azure Support.  I will let you know once i hear back from them.


I'm using visual studio code versions are below but both failed for me (MSAL and ADAL).  It looks like the same error. 

<img width="556" alt="image" src="https://user-images.githubusercontent.com/601055/224482303-644ac3c6-bfc4-406b-9368-4b8768c72881.png">

<img width="555" alt="image" src="https://user-images.githubusercontent.com/601055/224482384-bb6ae1d1-5a05-451e-a5d6-44be5e55dfc9.png">

=

VS Code Version:

Version: 1.76.1
Commit: 5e805b79fcb6ba4c2d23712967df89a089da575b
Date: 2023-03-08T16:32:09.831Z (2 days ago)
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Darwin x64 22.3.0
Sandboxed: No

**Comment by cpirtea-bun** [comment]
> > Hi everyone,
> > I've learned from an internal thread, [v1.39.1](https://github.com/microsoft/azuredatastudio/releases/tag/1.39.1) of ADS seems to work in this use-case, can someone who's affected confirm the same? Also please try with [v1.40.2](https://github.com/microsoft/azuredatastudio/releases/tag/1.40.2) as that would be useful for us to diagnose the problem. If it does, you may use the same to stay unblocked, while we continue with investigations.
> 
> 1.39.1 works, but 1.40.2 does not. The error logs from 1.40.2 are:
> 

Same thing here. 1.39.1 works, 1.40.2 does not.

**Comment by ncapito** [comment]
@cheenamalhotra  i found another work around.  I disabled this setting "System Certificates".

![image](https://user-images.githubusercontent.com/601055/224786834-8c746516-2339-4ce8-a120-de3069e9c60e.png)

Once i did that i was able to authorize using ADAL and MSL

**Comment by cheenamalhotra** [comment]
Interesting! Could you also test with the latest insiders build and let us know if that helps in this case?

**Comment by PhilHannent** [comment]
Good morning, I can confirm that unticking the system certificates solved the issue in both Production and the latest Insider build. Reticking the box causes the issue to come back.

Version: 1.43.0-insider
Commit: ef99e67cfe16ba993ccec2de266c5f973492e5de
Date: 2023-03-14T05:34:10.238Z
VS Code: 1.67.0
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Darwin arm64 22.3.0

and

Version: 1.41.2
Commit: 0f0960d0e774e1ce903b79378addff96117c27a6
Date: 2023-02-08T20:15:24.215Z
VS Code: 1.67.0
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Darwin x64 22.3.0

**Comment by StacyCMay** [comment]
Confirmed here as well. Unchecking `System Certificates` setting works in 1.41.2

**Comment by ncapito** [comment]
> @ncapito if possible, it would be helpful to have an Azure Support case for this, so we can get details on the user identities and times of the auth attempts (to try to correlate with errors in the service telemetry). And to be able to request more detailed logs that we can't collect over GitHub due to potential for PII data.

What would you like me to ask Azure to do?  They are asking me to do some random stuff, i would like to help guide the convo.

**Comment by cheenamalhotra** [comment]
Hi @ncapito 

It does seem like a certificate is involved when this error occurs, if you could investigate which certificate is contributing to this error with Azure support that would be helpful.

**Comment by johnborges** [comment]
Unchecking Http: System Certs and switching to ADAL default connection works for me.

**Comment by cheenamalhotra** [comment]
It's strange, since others were able to unblock with MSAL too.. 

Could you also test adding account with [VSCode-MSSQL extension](https://github.com/microsoft/vscode-mssql/releases/tag/v1.18.0) in VS Code as well? You can find Azure related logs in the bottom Output pane: **Azure Logs** when using extension.

![image](https://user-images.githubusercontent.com/13396919/227289992-830a1b2c-dca4-40ba-84b6-9dc68b4e3847.png)

Please attach any errors that you notice when adding account.

**Comment by cheenamalhotra** [comment]
Hi Everyone,

We were able to find the expired Trusted Root Certificate Authority that caused this issue with one of the affected customers.

1. You can navigate to: "Manage Computer Certificates" > Trusted Root Certification Authorities > Certificates
2. Find the DigiCert certificate expired on March 8, 2023, e.g. "DigiCert SHA2 Secure Server CA"

It's likely that due to organization policies this certificate wasn't updated in your case. You can find more info in the article from DigiCert: https://knowledge.digicert.com/generalinformation/digicert-root-and-intermediate-ca-certificate-updates-2023.html (March 8, 2023, ICA/Root Replacements) - that explains how the issue started on March 8, 2023.

You would likely need to install the updated DigiCert Trusted Root CA Certificates. Please contact your organization or DigiCert support for official guidance on the same.

**Comment by cheenamalhotra** [comment]
I will proceed to close the issue as external.

**Comment by ncapito** [comment]
@cheenamalhotra  I work somewhere where we have relatively loose controls around this.  I've experience this in Azure Data Studio and VS code (today).  I'm not understanding how this is an organizational issue.  Do you have instructions or guidance on how i can check this on a mac?
